app/jobs/regular/emit_web_hook_event.rb

```diff
@@ -3,45 +3,80 @@
 module Jobs
   class EmitWebHookEvent < Jobs::Base
     def execute(args)
-      raise Discourse::InvalidParameters.new(:web_hook_id) unless args[:web_hook_id].present?
-      raise Discourse::InvalidParameters.new(:event_type) unless args[:event_type].present?
-
-      args = args.dup
-
-      if args[:topic_id]
-        args[:topic_view] = TopicView.new(args[:topic_id], Discourse.system_user)
+      [:web_hook_id, :event_type].each do |key|
+        raise Discourse::InvalidParameters.new(key) unless args[key].present?
       end
 
-      if args[:post_id]
-        # deleted post so skip
-        return unless args[:post] = Post.find_by(id: args[:post_id])
-      end
+      web_hook = WebHook.find_by(id: args[:web_hook_id])
+      raise Discourse::InvalidParameters(:web_hook_id) if web_hook.blank?
 
-      if args[:user_id]
-        return unless args[:user] = User.find_by(id: args[:user_id])
-      end
-
-      web_hook = WebHook.find(args[:web_hook_id])
-
-      unless args[:event_type] == 'ping'
+      unless ping_event?
         return unless web_hook.active?
+
         return if web_hook.group_ids.present? && (args[:group_id].present? ||
           !web_hook.group_ids.include?(args[:group_id]))
+
         return if web_hook.category_ids.present? && (!args[:category_id].present? ||
           !web_hook.category_ids.include?(args[:category_id]))
+
+        event_type = args[:event_type].to_s
+        return unless self.send("setup_#{event_type}")
       end
 
       web_hook_request(args, web_hook)
     end
 
     private
 
-    def web_hook_request(args, web_hook)
+    def guardian
+      Guardian.new(Discourse.system_user)
+    end
+
+    def setup_post(args)
+      post = Post.find_by(id: args[:post_id])
+      raise Discourse::InvalidParameters.new(:post_id) if post.blank?
```

Comment:
Sam made a change months ago about not raising exceptions for post. There could be many deleted posts.

Ref: https://github.com/discourse/discourse/commit/ac2c035856ad0cdb2ca7a3b5ccf7df18e15a2914